app/src/main/java/free/rm/skytube/gui/fragments/PreferencesFragment.java

```diff
@@ -104,10 +113,46 @@ public void onPause() {
 
 	@Override
 	public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
-		// If the user changed the Default Tab Preference, update the summary to show the new default tab
-		if(key != null && key.equals(getString(R.string.pref_key_default_tab))) {
-			ListPreference defaultTabPref = (ListPreference)findPreference(key);
-			defaultTabPref.setSummary(String.format(getString(R.string.pref_summary_default_tab), defaultTabPref.getEntry()));
+		if(key != null) {
+			if (key.equals(getString(R.string.pref_key_default_tab))) {
+				// If the user changed the Default Tab Preference, update the summary to show the new default tab
+				ListPreference defaultTabPref = (ListPreference) findPreference(key);
+				defaultTabPref.setSummary(String.format(getString(R.string.pref_summary_default_tab), defaultTabPref.getEntry()));
+			} else if (key.equals(getString(R.string.pref_youtube_api_key))) {
+				// Validate the entered API Key when saved (and not empty), with a simple call to get the most popular video
+				final EditTextPreference youtubeAPIKeyPref = (EditTextPreference) findPreference(getString(R.string.pref_youtube_api_key));
+				if(!youtubeAPIKeyPref.getText().isEmpty()) {
+					new AsyncTask<Void, Void, Boolean>() {
+						@Override
+						protected Boolean doInBackground(Void... voids) {
+							try {
+								YouTube.Videos.List videosList = YouTubeAPI.create().videos().list("snippet");
+								videosList.setFields("items(id)");
+								videosList.setKey(youtubeAPIKeyPref.getText());
+								videosList.setChart("mostPopular");
+								String regionCode = SkyTubeApp.getPreferenceManager().getString(SkyTubeApp.getStr(R.string.pref_key_preferred_region), "").trim();
+								videosList.setRegionCode(regionCode.isEmpty() ? null : regionCode);
```

Comment:
Also, if a user has previously entered an API Key and taps on "YouTube API Key", i would like that in that case, the old key is displayed in the text box.